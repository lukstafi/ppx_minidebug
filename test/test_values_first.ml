(* Simple test to demonstrate values_first_mode *)

let _get_local_debug_runtime =
  let rt = Minidebug_db.debug_db_file ~elapsed_times:Microseconds "test_values_first" in
  fun () -> rt

let%debug_show rec factorial (n : int) : int = if n <= 1 then 1 else n * factorial (n - 1)
let%debug_show add (x : int) (y : int) : int = x + y

let%debug_show compute (a : int) (b : int) : int =
  let sum : int = add a b in
  factorial sum

let () =
  let _ = compute 3 2 in
  let db = Minidebug_client.Client.open_db "test_values_first.db" in
  Minidebug_client.Client.show_trace db ~values_first_mode:true;
  ()
