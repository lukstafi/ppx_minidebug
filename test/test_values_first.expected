[debug] compute => 120 @ test/test_values_first.ml:10:23-12:15
  a = 3
  b = 2
  [debug] sum => 5 @ test/test_values_first.ml:11:6-11:9
    [debug] add => 5 @ test/test_values_first.ml:8:19-8:52
      x = 3
      y = 2
  [debug] factorial => 120 @ test/test_values_first.ml:7:29-7:90
    n = 5
    [debug] factorial => 24 @ test/test_values_first.ml:7:29-7:90
      n = 4
      [debug] factorial => 6 @ test/test_values_first.ml:7:29-7:90
        n = 3
        [debug] factorial => 2 @ test/test_values_first.ml:7:29-7:90
          n = 2
          [debug] factorial => 1 @ test/test_values_first.ml:7:29-7:90
            n = 1
